### ISSUE TITLE
R std_vector.i not wrapping std::vector<unsigned long int> with SWIGWORDSIZE64 defined

### ISSUE THREAD

**Issue by blowekamp** [swig/swig#2847]
This is a problem detected when updating from SWIG 4.1.1 to 4.2.1. A git bisect on SWIG reveals commit  9fb3a4939e4ec528f050057d8ccd743a066222ac was the cause of the regression.

A minimal case is something like:

```
%module test

%include <std_vector.i>

# might not be needed
%template(VectorUInt64) vector<uint64_t>;
%inline {

  std::vector<uint64_t> vu64(void) { return std::vector<uint64_t>{1, 2}; }

}
```

The return values for this type is a raw pointer e.g. `pointer: 0x561e664b81c0>` when printed.

Adding "-DSWIGWORDSIZE64" to compile command line the cxx file resolves the issue.

The [Lib/r/std_vector.i](https://github.com/swig/swig/blob/master/Lib/r/std_vector.i#L126) contains `#if defined(SWIGWORDSIZE64)`, that are making it into the cxx file, but I suspect were intended to be SWIG preprocessor directives and not C++. 

Can the preprocess definitions just be removed? Can both `std::vector<long int>` and `std::vector<long long>` be wrapped if needed?

**Comment by ojwb** [comment]
Thanks for the analysis, so the problem looks to be caused by removal by that commit of the line:

```
#define SWIGWORDSIZE64
```

@wsfulton That change doesn't seem to be covered by the commit message so maybe this wasn't intended to be included?

**Comment by wsfulton** [comment]
The way this should work is by default `long` is treated as 32-bit and then if SWIGWORDSIZE64 is defined then `long` is treated as 64-bit.

But in R, only `vector<long long>` is provided by default and `vector<long>` is missing, and when SWIGWORDSIZE64 is defined  `vector<long>` is provided and `vector<long long>` is missing.

I propose to fix this by always defining `vector<long long>` and always defining `vector<long>.

The implementations for `int`, `long` and `long long` are all identical in https://github.com/swig/swig/blob/21aa6e8f9e4c7d1b0249c46140b7ce5397d7dc52/Lib/r/std_vector.i#L110-L182. So in conclusion, I will remove the SWIGWORDSIZE64 macro altogether in this file.

When done SWIGWORDSIZE64 won't be needed at all.

**Comment by wsfulton** [comment]
Turns out that you do need `-DSWIGWORDSIZE64` as you must be using stdint.i in order to get the definition of `uint64_t`. Here is a fixed example that works after my 8f2a4af906517a08dec947b08e28cd373ef5e7b4 commit today (assuming you are using 64-bit long of course):

```swig
%module example

#define SWIGWORDSIZE64

%include <stdint.i>
%include <std_vector.i>

%template(VectorUInt64) std::vector<uint64_t>;
%inline {
  std::vector<uint64_t> vu64(void) { return std::vector<uint64_t>{1, 2}; }
}
```

**Comment by blowekamp** [comment]
Great, I'll test it out. And this is only going to be available in 4.3? as there is no planned 4.2.2?

**Comment by wsfulton** [comment]
> Great, I'll test it out. And this is only going to be available in 4.3? as there is no planned 4.2.2?

Currently, the next planned release is 4.3 and no 4.2.2 is planned. I'd consider issuing a 4.2.2 if there were a number of regressions to fix that are hard to work around, but that is not the case at the moment.

**Comment by blowekamp** [comment]
Thank you. I'll add the if "4.2.x" checks around the additional define.
